### PR TITLE
Various improvements to AdaptVQE

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

This PR adds a few minor improvements to the `AdaptVQE` implementation.
Mostly more logging to make it easier to follow the progress of the algorithm.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
Could we include testing the eigenvalue_history, maybe in one (or more) of the existing tests, to ensure its there in the result as expected.

**Comment by mrossinek** [comment]
> Could we include testing the eigenvalue_history, maybe in one (or more) of the existing tests, to ensure its there in the result as expected.

Will do that next week :+1: 

I was actually also thinking just now, that we could also store the `gradient` history as this might be useful in post-processing, too

File: releasenotes/notes/adapt-vqe-improvements-8617aaa94a6e6621.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    Adds the :attr:`.AdaptVQEResult.eigenvalue_history` attribute.
```
Comment (Cryoris) [inline releasenotes/notes/adapt-vqe-improvements-8617aaa94a6e6621.yaml:4]:
```suggestion
    Adds the :attr:`.AdaptVQEResult.eigenvalue_history` attribute. This tracks the lowest
    achieved energy per iteration of the AdaptVQE.
```

**Review by Cryoris — APPROVED** [review]

**Comment by ikkoham** [comment]
@Mergifyio refresh

**Comment by Cryoris** [comment]
@mergifyio requeue

**Pull request merged**